TACC/protx
# Add first step at providing maltreatment data from sqlite source

## Overview: ##

This PR begins to use the sqlite database as a data source instead of preprocessed json files.

## Related Jira tickets: ##

* [COOKS-118](https://jira.tacc.utexas.edu/browse/COOKS-118)
* [COOKS-24](https://jira.tacc.utexas.edu/browse/COOKS-24)


## Summary of Changes: ##
* Use maltreatment data from sqlite database
* Place database and static files (vector, json) in specific volume

## Testing Steps: ##
0. Get data (**see below in NOTES**)
1. Ensure that you built base image as there is a new python requirement (`make build`)
2. Look at `/protx/maltreatment/` and ensure maltreatment view remains the same
3. Check some other views views

## UI Photos:

None.


## Notes: ##

Get data with the following
```
mkdir ~/protx-data
mv  PROTX_REPO_LOCAL_DEV/server/cms/static/data ~/protx-data/static
cp test_20210907.db ~/protx-data/cooks.db 
```


TODO
- [x] missing 2010 data (Kelly is looking into that in https://jira.tacc.utexas.edu/browse/COOKS-131)
- [x] need to have /data directory where DB (and where other map data can eventually move to)
- [x] add reminder in code about requiring login for the maltreatment route in the future. 
- [x] update Core-Portal-Deployments for these docker volumes and nginx overrides
- [x] add data to staging
- [ ] add database to prod will have to be done before a future prod deployment
